nuxeo-features/nuxeo-automation/nuxeo-automation-features/src/test/java/org/nuxeo/ecm/automation/core/operations/services/bulk/TestSetPropertiesActionOptimizations.java

```diff
@@ -0,0 +1,188 @@
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.ecm.core.test.DefaultRepositoryInit;
+import org.nuxeo.ecm.core.test.annotations.RepositoryConfig;
+import org.nuxeo.ecm.platform.audit.AuditFeature;
+import org.nuxeo.ecm.platform.audit.api.AuditLogger;
+import org.nuxeo.ecm.platform.audit.service.NXAuditEventsService;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * Test audit and/or auto versioning disablement.
+ * <p>
+ * It's done at this level to have those features loaded.
+ *
+ * @since 10.3
+ */
+@RunWith(FeaturesRunner.class)
+@Features({ CoreFeature.class, AuditFeature.class })
+@RepositoryConfig(init = DefaultRepositoryInit.class)
+@Deploy("org.nuxeo.ecm.core.test.tests:OSGI-INF/test-repo-core-types-contrib.xml")
+public class TestSetPropertiesActionOptimizations {
+
+    @Inject
+    public CoreSession session;
+
+    @Inject
+    public BulkService bulkService;
+
+    @Inject
+    public NXAuditEventsService auditService;
+
+    @Inject
+    public TransactionalFeature transactions;
+
+    @Inject
+    public VersioningService versioningService;
+
+    @Inject
+    public EventService eventService;
+
+    @Inject
+    public AuditLogger auditLogger;
+
+    protected DocumentModel note;
+
+    private class TestListener extends EventListenerDescriptor {
+
+        private int count;
+
+        private TestListener() {
+            events = new HashSet<>(auditLogger.getAuditableEventNames());
+        }
+
+        @Override
+        public void initListener() {
+            inLineListener = event -> count++;
+        }
+    }
+
+    @Before
+    public void before() {
+        DocumentModel workspace = session.getDocument(new PathRef("/default-domain/workspaces/test"));
+        DocumentModel note = session.createDocumentModel(workspace.getPathAsString(), "note", "Note");
+        this.note = session.createDocument(note);
+        transactions.nextTransaction();
+    }
+
+    @Test
+    public void testDisableAudit() throws InterruptedException {
+        TestListener listener = new TestListener();
+        eventService.addEventListener(listener);
+        assertEquals(0, listener.count);
+
+        // should increment listener count by 3
+        // documentCheckedIn, documentCreated and documentModified are fired
+        bulkService.submit(createBuilder().build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        assertEquals(3, listener.count);
+
+        // should increment listener count by 1, as documentModified is not avoidable
+        // only documentModified is fired
+        bulkService.submit(createBuilder().param(PARAM_DISABLE_AUDIT, TRUE).build());
+        assertTrue("Bulk action didn't finish", bulkService.await(Duration.ofSeconds(60)));
+        assertEquals(4, listener.count);
```

Comment:
Here we should hit automatic versioning no ? We should have a checkout, then a checkin and a document created. Let's discuss